security/encryption.go

```diff
@@ -60,40 +81,60 @@ func (cd *aesGcmCrypting) validate() error {
 	return errors.New(errorsMsg.MsgErrUnknown + "." + errorsMsg.EncryptDecrypt)
 }
 
-func (cd *aesGcmCrypting) Encrypt(value []byte, additional []byte) ([]byte, error) {
-	var block, err = aes.NewCipher(cd.key)
+func (km *keyMaterial) Encrypt(value []byte, additional []byte) ([]byte, string, error) {
+	// select the most recent key
+	key := km.Keys[0]
+	var block, err = aes.NewCipher(key.Key)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
 	var iv = make([]byte, 12)
 	_, _ = rand.Read(iv)
 
 	var aesgcm cipher.AEAD
-	aesgcm, err = cipher.NewGCMWithTagSize(block, cd.tagSize)
+	aesgcm, err = cipher.NewGCMWithTagSize(block, key.tagSize)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
 	var enc = aesgcm.Seal(nil, iv, value, additional)
+	kid := key.Kid
+	encValue := append(iv, enc...)
 
-	return append(iv, enc...), err
+	return encValue, kid, err
```

Comment:
I would suggest to not return the kid here and add a methods to getthe default kid.

Reply:
ok, let's do that :-)